File: java-pi-pantry/src/test/java/com/langrsoft/util/AnHttpClient.java
package com.langrsoft.util;

import com.langrsoft.testutil.Slow;
import org.junit.Test;
import org.junit.experimental.categories.Category;

import static org.hamcrest.Matchers.containsString;
import static org.junit.Assert.assertThat;

public class AnHttpClient {
    @Category(Slow.class)
    @Test
    public void canRetrieveTextResponseFromUrl() {
        HttpClient client = new HttpClient();
        String teapotStatus = "418";
        String text = client.retrieveText("http://httpbin.org/status/" + teapotStatus);

        assertThat(text.toLowerCase(), containsString("teapot"));
    }
}

File: csharp-pi-pantry/csharp-pi-pantry-test/UtilTests/AnHttpClient.cs
using NUnit.Framework;
using Pipantry.Util;
using System;

namespace Test.Pipantry.Util
{
    [TestFixture, Category("slow")]
    public class AnHttpClient
    {
        const int TeapotStatus = 418;
        private Http client;

        [SetUp]
        public void Create()
        {
            client = new Http();
        }

        [Test]
        public void CanRetrieveTextResponseFromUrl()
        {
            var text = client.RetrieveText("http://httpbin.org/ip");

            Assert.That(text.ToLower(), Does.Contain("\"origin\""));
        }

        [Test]
        public void ThrowsExceptionOnFailure()
        {
            Assert.Throws<ApplicationException>(() => client.RetrieveText("http://httpbin.org/status/{teapotStatus}"));
        }

        [Test]
        public void AccommodatesCustomErrorHandler()
        {
            var text = client.RetrieveText($"http://httpbin.org/status/{TeapotStatus}", status => $"status: {status}");

            Assert.That(text, Is.EqualTo("status: 418"));
        }
    }
}
